File: Python_Codes/14. Object Class Methods/GetClassExample.py
# Class representing a custom data type
class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age

# Main logic to demonstrate class type retrieval
def main():
    person = Person("John Doe", 30)
    string_object = "Hello, World!"
    integer_object = 42

    # Retrieve and print the runtime class of each object
    print("Class of person object:", person.__class__.__name__)
    print("Class of stringObject:", string_object.__class__.__name__)
    print("Class of integerObject:", integer_object.__class__.__name__)

if __name__ == "__main__":
    main()

File: Projects/JavaProject/src/com/Object_Class_Methods/GetClassExample.java
package com.Object_Class_Methods;

// Class representing a custom data type
class Person {
    // Constructor to initialize the Person object
    public Person(String name, int age) {
    }
}

// Main class to demonstrate the getClass() method
public class GetClassExample {
    public static void main(String[] args) {
        // Create objects of different classes
        Person person = new Person("John Doe", 30);
        String stringObject = "Hello, World!";
        Integer integerObject = 42;

        // Retrieve and print the runtime class of each object
        System.out.println("Class of person object: " + person.getClass().getName());
        System.out.println("Class of stringObject: " + stringObject.getClass().getName());
        System.out.println("Class of integerObject: " + integerObject.getClass().getName());
    }
}
